### PR TITLE
Notifications support

### LINKED ISSUE #23: Wallet login based on login.xyz
Adapt the current signin/login experience to use EIP-4361 messages with https://docs.login.xyz/

### LINKED ISSUE #46: Webhook triggers for on-chain events

### PR DESCRIPTION
**Pull request by opny721**
This PR brings back notifications support. 

Note: once in staging it should be tested by someone with >5% stake to confirm it is working correctly, thanks

Also, authentication has been migrated to wagmi and siwe messaging/signature system

closes #88 
closes #46 
closes #23

### CODE REVIEW

**Comment by ivelin** [comment]
After merge with latest dev branch, I expected to see a prompt from SIWE but instead I see Auth error message with a spinning error log in dev console. See screenshots below.

The Auth error message only shows up when I go to Settings page. No error on Home, People and Projects page. Testing with "PM Test DAO" on Goerli.

`/dao/chain/5/address/0xe237747055b12f4da323bc559ac8d5eb66aac2f7/projects`

Screen Shot 2022-07-23 at 3 37 21 PM

Screen Shot 2022-07-23 at 3 32 30 PM

Screen Shot 2022-07-23 at 3 32 24 PM

File: .github/workflows/push-image.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+name: Publish Docker image
```
Comment (ivelin) [inline .github/workflows/push-image.yaml:1]: Looks good. Should we tie this with semrel or some other release pipeline?

Comment (muka) [reply .github/workflows/push-image.yaml:1]: Yup I am very new to GH actions if you have some hint would be really appreciated

Comment (ivelin) [reply .github/workflows/push-image.yaml:1]: See this recent blog on semrel with gh actions:
https://levelup.gitconnected.com/using-semantic-release-with-github-actions-c30d197829f1

We've been using semrel for docker hub releases on ambianic edge images:
https://github.com/ambianic/ambianic-edge/blob/master/.github/workflows/main.yml

Comment (opny721) [reply .github/workflows/push-image.yaml:1]: Ok will check and fix merge issues

**Comment by ivelin** [comment]
Awesome! Reviewing today.

**Comment by ivelin** [comment]
Got his message on startup:
Screen Shot 2022-09-01 at 1 32 09 PM

**Comment by ivelin** [comment]
I am able to build and run. Settings for webhook page show up. But I still hit a wall with these Auth errors and the form does not seem to be able to save. No obvious errors in the `make` log. See screenshots below.

```
docker-compose up -d mongo
make prod/build prod/run
# go to localhost:3001
```

Screen Shot 2022-09-01 at 1 41 17 PM
Screen Shot 2022-09-01 at 1 42 45 PM

**Comment by ivelin** [comment]
Screen Shot 2022-09-01 at 1 45 01 PM

**Comment by opny721** [comment]
Ok, if you start the local image just go to `http://localhost:3001` so you get the FE+BE in one place

I have not tried the gitpod deployment yet, I will look into it now

**Comment by opny721** [comment]
Fixed now, there was a double rendering causing the signin nonce to  be regenerated

**Comment by ivelin** [comment]
> Fixed now, there was a double rendering causing the signin nonce to be regenerated

Awesome! Testing...

**Comment by ivelin** [comment]
When I try to start a gitpod instance on this branch, it seems to hang up. It has been in starting state for 20 minutes now. Usually takes no more than 1 minute to launch a new workspace.  How quick does it launch for you, @opny721 ?

https://gitpod.io/#https://github.com/SporosDAO/sweat-token/pull/95


Screen Shot 2022-09-02 at 11 16 26 AM


Screen Shot 2022-09-02 at 11 17 08 AM

**Comment by ivelin** [comment]
For comparison, gitpod launches in about 1 min on the latest dev branches and my fork:
https://gitpod.io/#https://github.com/SporosDAO/sweat-token
https://gitpod.io/#https://github.com/ivelin/sweat-token

**Comment by ivelin** [comment]
Screen Shot 2022-09-02 at 2 36 19 PM

**Comment by opny721** [comment]
Also about a minute. Maybe a transient issue ?

image

**Comment by ivelin** [comment]
@opny721 You were right. The gitpod blocking issue went away. I am now able to open the app in gitpod and start it.

A couple new issues popped up:
1. There seem to be a db init issue. Authentication fails when I try to sign with the dev key on Goerli that I normally use. 
2. The message asking the user to sign is a mystery. Can we display a more user friendly text that explains to the user what they are signing?

See screenshots below:

Screen Shot 2022-09-06 at 9 56 55 AM

Screen Shot 2022-09-06 at 9 57 21 AM
Screen Shot 2022-09-06 at 9 57 59 AM
Screen Shot 2022-09-06 at 10 01 40 AM
Screen Shot 2022-09-06 at 10 02 06 AM
Screen Shot 2022-09-06 at 10 02 23 AM

**Comment by ivelin** [comment]
@opny721 should I close this PR for now? Or you think you will have time to work on it. 
I am trying to cleanup the product board for the end of October release.

**Comment by ivelin** [comment]
Closing as stale for now to clean up the PR board. We can reopen when work resumes.

**Pull request closed without merging**